eslint.config.js

```diff
@@ -0,0 +1,316 @@
+import jqueryConfig from "eslint-config-jquery";
+import importPlugin from "eslint-plugin-import";
+import globals from "globals";
+
+export default [
+	{
+
+		// Only global ignores will bypass the parser
+		// and avoid JS parsing errors
+		// See https://github.com/eslint/eslint/discussions/17412
+		ignores: [
+			"test/data/json_obj.js"
+		]
+	},
+
+	{
+		files: [
+			"eslint.config.js",
+			"Gruntfile.cjs",
+			"test/node_smoke_tests/commonjs/**",
+			"test/node_smoke_tests/module/**",
+			"test/promises_aplus_adapters/**",
+			"test/middleware-mockserver.cjs"
+		],
+		languageOptions: {
+			globals: {
+				...globals.node,
+				...globals.es2021
+			}
+		},
+		rules: {
+			...jqueryConfig.rules,
+			strict: [ "error", "global" ]
+		}
+	},
+
+	// Source
+	{
+		files: [ "src/**" ],
```

Comment:
`main` has this comment for browser files above an empty `env`:
```js
	// The browser env is not enabled on purpose so that code takes
	// all browser-only globals from window instead of assuming
	// they're available as globals. This makes it possible to use
	// jQuery with tools like jsdom which provide a custom window
	// implementation.
    env: {},
```

I think it's worth preserving it somewhere here.